src/ja/server/main.py

```diff
@@ -32,21 +32,24 @@ def _read_config() -> ServerConfig:
         with open('/etc/jobadder/server.conf') as f:
             return ServerConfig.from_string(f.read())
 
-    def __init__(self) -> None:
+    def __init__(self, socket_path: str = "/run/jobadder-server.socket", database_name: str = "jobadder") -> None:
         """!
         Initialize the JobAdder server daemon.
         This includes:
         1. Parsing the command line arguments and the configuration file.
         2. Connecting to the configured database.
         3. Initializing the scheduler and the dispatcher.
         4. Starting the web server and the email notifier.
+        @param socket_path: the path to the unix named socket for the command handler to listen on.
+        @param database_name: the name of the database to use.
         """
 
         config = self._read_config()
         self._database = SQLDatabase(host=config.database_config.host,
                                      user=config.database_config.username,
-                                     password=config.database_config.password)
-        proxy_factory = WorkerProxyFactory(self._database)
+                                     password=config.database_config.password,
+                                     database_name=database_name)
+        proxy_factory = self._get_proxy_factory()
```

Comment:
I thought the goal is to test with all real components, including the real proxy factory. Why do we need to change that?

Reply:
Hmm, is it because you want to use Proxies which use the custom Remotes ?

Reply:
The default locations for configs and sockets are only accessible by root.
I added subclasses of the main classes and some other classes that define different paths for configs and sockets.
This is to allow non-root users to run the integration tests and to prevent unintentional changes to the system (just imagine a bug like "rm -r /etc /jobadder" instead of "rm -r /etc/jobadder").

Reply:
Yeah, makes sense.